src/assets/LESSAsset.js

```diff
@@ -80,17 +83,65 @@ function getFileManager(less, options) {
       return false;
     }
 
-    async loadFile(filename, currentDirectory) {
-      filename = parseCSSImport(filename);
-      let resolved = await resolver.resolve(
+    async loadFile(filename, currentDirectory, options) {
+      let resolved = await this.findResolution(
         filename,
-        path.join(currentDirectory, 'index')
+        currentDirectory,
+        options
       );
+
       return {
         contents: await fs.readFile(resolved.path, 'utf8'),
         filename: resolved.path
       };
     }
+
+    async findResolution(filename, currentDirectory, options) {
+      const filenamesTried = [];
+      let paths = path.isAbsolute(filename) ? [] : [currentDirectory];
+
+      if (options.paths) {
+        paths = paths.concat(options.paths);
+      }
+
+      if (!path.isAbsolute(filename) && !paths.includes('.') === -1) {
+        paths.push('.');
+      }
+
+      // promise is guaranteed to be asyncronous
+      // which helps as it allows the file handle
+      // to be closed before it continues with the next file
+      return new Promise(function(resolve, reject) {
+        (async function tryPathIndex(i) {
```

Comment:
You can probably split this up into a seperate function and await it instead of wrapping it in a promise in a IFFE.
It would make the code more readable and probably smaller

Something like this
```Javascript
const tryPathIndex = async i => {
  // the code of the function...
}
return await tryPathIndex(0);
```